JUnit Pioneer v1 issues

> There doesn't seem to be the substantial drop in coverage that was previously seen with the proposed bump to v1.0.0 in #2846 -- so it looks like this might be good to go!
> 
> Cross-referencing / links:
> 
> #2846
> #2891
> #2893
> 
> [#2846 (comment)](https://github.com/javaparser/javaparser/pull/2846#issuecomment-723513550)
> [junit-pioneer/junit-pioneer#343](https://github.com/junit-pioneer/junit-pioneer/issues/343)

Looks like I was wrong -- I'll revert the commit shortly.

Specifically, note that this update causes the tests to fail, but maven reports a pass:

``` 
[INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ javaparser-symbol-solver-testing ---
[INFO] Surefire report directory: /home/runner/work/javaparser/javaparser/javaparser-symbol-solver-testing/target/surefire-reports
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
Mar 07, 2021 3:37:38 PM org.junit.platform.launcher.core.DefaultLauncher handleThrowable
WARNING: TestEngine with ID 'junit-jupiter' failed to execute tests
java.lang.NoClassDefFoundError: org/junit/platform/commons/util/UnrecoverableExceptions
	at org.junit.jupiter.engine.support.OpenTest4JAndJUnit4AwareThrowableCollector.createAbortedExecutionPredicate(OpenTest4JAndJUnit4AwareThrowableCollector.java:58)
	at org.junit.jupiter.engine.support.OpenTest4JAndJUnit4AwareThrowableCollector.<clinit>(OpenTest4JAndJUnit4AwareThrowableCollector.java:41)
	at org.junit.jupiter.engine.support.JupiterThrowableCollectorFactory.createThrowableCollector(JupiterThrowableCollectorFactory.java:34)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:78)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:229)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:197)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:191)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:188)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:128)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
Caused by: java.lang.ClassNotFoundException: org.junit.platform.commons.util.UnrecoverableExceptions
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:582)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:185)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:496)
	... 19 more

[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
Warning:  Corrupted STDOUT by directly writing to native stream in forked JVM 1. See FAQ web page and the dump file /home/runner/work/javaparser/javaparser/javaparser-symbol-solver-testing/target/surefire-reports/2021-03-07T15-36-16_653-jvmRun1.dumpstream
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.6:report (jacoco-report) @ javaparser-symbol-solver-testing ---
[INFO] Skipping JaCoCo execution due to missing execution data file.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for javaparser-parent 3.19.1-SNAPSHOT:
[INFO] 
[INFO] javaparser-parent .................................. SUCCESS [  0.277 s]
[INFO] javaparser-core .................................... SUCCESS [ 15.871 s]
[INFO] javaparser-core-testing ............................ SUCCESS [ 59.483 s]
[INFO] javaparser-core-testing-bdd ........................ SUCCESS [ 12.829 s]
[INFO] javaparser-core-generators ......................... SUCCESS [  1.153 s]
[INFO] javaparser-core-metamodel-generator ................ SUCCESS [  0.156 s]
[INFO] javaparser-core-serialization ...................... SUCCESS [  1.309 s]
[INFO] javaparser-symbol-solver-core ...................... SUCCESS [  1.807 s]
[INFO] javaparser-symbol-solver-testing ................... SUCCESS [  5.442 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:38 min
[INFO] Finished at: 2021-03-07T15:37:38Z
[INFO] ------------------------------------------------------------------------
```

_Originally posted by @MysterAitch in https://github.com/javaparser/javaparser/issues/3121#issuecomment-792304197_

--- a/javaparser-symbol-solver-testing/pom.xml
+++ b/javaparser-symbol-solver-testing/pom.xml
@@ -185,10 +185,6 @@
             <artifactId>junit-jupiter-params</artifactId>
         </dependency>
         <dependency>
-            <groupId>org.junit-pioneer</groupId>
-            <artifactId>junit-pioneer</artifactId>
-        </dependency>
-        <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-core</artifactId>
         </dependency>

--- a/pom.xml
+++ b/pom.xml
@@ -375,13 +375,6 @@
                 <scope>test</scope>
             </dependency>
             <dependency>
-                <groupId>org.junit-pioneer</groupId>
-                <artifactId>junit-pioneer</artifactId>
-                <!-- TODO: Not yet upgrading to v1.0.0 (causes some tests to not execute - see #2891 and #2893) -->
-                <version>0.5.6</version>
-                <scope>test</scope>
-            </dependency>
-            <dependency>
                 <groupId>org.mockito</groupId>
                 <artifactId>mockito-core</artifactId>
                 <version>3.8.0</version>

